setup.py

```diff
@@ -77,11 +77,16 @@ def build_extension(self, ext: CMakeExtension) -> None:
         # Pass version to C++
         cmake_args += [f"-DMLX_VERSION={self.distribution.get_version()}"]  # type: ignore[attr-defined]
 
-        if sys.platform.startswith("darwin"):
+        if platform.system() == "Darwin":
             # Cross-compile support for macOS - respect ARCHFLAGS if set
             archs = re.findall(r"-arch (\S+)", os.environ.get("ARCHFLAGS", ""))
             if archs:
                 cmake_args += ["-DCMAKE_OSX_ARCHITECTURES={}".format(";".join(archs))]
+        if platform.system() != "Windows":
+            # On Windows DLLs must be put in the same dir with the extension
+            # while cmake puts mlx.dll into the "bin" sub-dir. Link with mlx
+            # statically to work around it.
+            cmake_args += ["-DBUILD_SHARED_LIBS=ON"]
```

Comment:
I had a hard time reading this, as it said "statically" and appended `-DBUILD_SHARED_LIBS=ON` (because I missed the `!=` in `!= "Windows"`.

Wouldn't it be more straightforward to have `-DBUILD_SHARED_LIBS=ON` in the default `cmake_args`, and then append an extra `-DBUILD_SHARED_LIBS=OFF` when `platform.system() == "Windows"`? (IIUC, the last cmake arg should take precedence, right?)

Reply:
Or more explicitly so readers don't have to check which arg takes precedence.

```python
if platform.system() = "Windows":
    cmake_args += ["-DBUILD_SHARED_LIBS=OFF"]
else:
    cmake_args += ["-DBUILD_SHARED_LIBS=ON"]
```

I'm not opposed to making the code more readable, I'm leaving the decision to @awni .

Reply:
I like the last option you suggest @zcbenz it is a lot easier to follow.

Reply:
Updated.